### PR TITLE
added securitygrouprule for accessing Elasticache/Redis

### PR DESCRIPTION
**Pull request by humoflife**
### Description of your changes

Added securitygrouprule for accessing Elasticache/Redis, so that this configuration can be used by a forthcoming configuration-aws-elasticache reference.

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

make e2e

### CODE REVIEW

File: examples/configuration.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: pkg.crossplane.io/v1
 kind: Configuration
 metadata:
-  name: cofiguration-aws-network
+  name: configuration-aws-network
 spec:
-  package: xpkg.upbound.io/upbound/configuration-aws-network:v0.1.0
+  package: xpkg.upbound.io/upbound/configuration-aws-network:v0.1.2
```
Comment (haarchri) [inline examples/configuration.yaml:6]: v0.13.0 after this PR

File: apis/basic/composition.yaml [hunk 2]
```diff
@@ -531,3 +531,28 @@ spec:
                 patchSetName: network-id
               - type: PatchSet
                 patchSetName: region
+
+          - name: securityGroupRuleRedis
+            base:
+              apiVersion: ec2.aws.upbound.io/v1beta1
+              kind: SecurityGroupRule
```
Comment (haarchri) [inline apis/basic/composition.yaml:538]: Can we start using the new Resource SecurityGroupIngressRule ? This resource using the real AWS identifier sgr-xxxxxx

Comment (humoflife) [reply apis/basic/composition.yaml]: updated, and also added SecurityGroupIngressRule for Memcached.

File: examples/configuration.yaml (hunk 1 above)
Comment (humoflife) [reply examples/configuration.yaml:6]: updated

File: examples/configuration.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 apiVersion: pkg.crossplane.io/v1
 kind: Configuration
 metadata:
-  name: cofiguration-aws-network
+  name: configuration-aws-network
 spec:
-  package: xpkg.upbound.io/upbound/configuration-aws-network:v0.1.0
+  package: xpkg.upbound.io/upbound/configuration-aws-network:v0.1.3
```
Comment (haarchri) [inline examples/configuration.yaml:6]: v0.13.0

Comment (haarchri) [reply examples/configuration.yaml:6]: @humoflife please check this

Comment (humoflife) [reply examples/configuration.yaml:6]: should this be v0.1.2 for now?

**Review by haarchri — APPROVED** [review]
please squash during merge

**Review by ytsarev — CHANGES_REQUESTED** [review]
@humoflife do you still need this change? If yes, please rebase with `main` and we can uptest it together with the new changes and eventually merge it. Thanks!

**Comment by ytsarev** [comment]
image
rebase does not look correct, I expect only your commit to be a part of this PR

**Comment by haarchri** [comment]
looks stale - @humoflife if you want this change please open a new PR

**Pull request closed without merging**